test/test_data.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import pandas as pd
+import pytest
+from anndata import AnnData
+
+from scvid.data import AnnDataSchema
+
+n_cell_ref, n_cell, n_gene = (3, 5, 4)
+
+
+@pytest.fixture
+def adata():
+    X = np.zeros((n_cell, n_gene))
+    L = np.zeros((n_cell, n_gene))
+    M = np.zeros((n_cell, 2))
+    V = np.zeros((n_gene, 2))
+    obs = pd.DataFrame(
+        dict(batch=np.array(["a", "b"])[np.random.randint(0, 2, n_cell)]),
+        index=[f"cell{i:03d}" for i in range(n_cell)],
+    )
+    var = pd.DataFrame(index=[f"gene{i:03d}" for i in range(n_gene)])
+    return AnnData(
+        X,
+        dtype=X.dtype,
+        obs=obs,
+        var=var,
+        layers={"L": L},
+        obsm={"M": M},
+        varm={"V": V},
+    )
+
+
+@pytest.fixture
+def ref_adata():
+    X = np.ones((n_cell_ref, n_gene))
+    L = np.ones((n_cell_ref, n_gene))
+    M = np.ones((n_cell_ref, 2))
+    V = np.zeros((n_gene, 2))
+    obs = pd.DataFrame(
+        dict(batch=np.array(["a", "b"])[np.random.randint(0, 2, n_cell_ref)]),
+        index=[f"ref_cell{i:03d}" for i in range(n_cell_ref)],
+    )
+    var = pd.DataFrame(index=[f"gene{i:03d}" for i in range(n_gene)])
+    return AnnData(
+        X,
+        dtype=X.dtype,
+        obs=obs,
+        var=var,
+        layers={"L": L},
+        obsm={"M": M},
+        varm={"V": V},
+    )
+
+
+def test_validate_adata(ref_adata, adata):
```

Comment:
What do you think about adding a couple of failing fixtures and test them for failure, for example:
- `obs` columns permuted
- `obs` column names are different
- `obs` columns names match but dtype of a column is different
- `obs` columns names match but the categories of a `pandas.Categorical` column are different
- `var` column names are in different order

Testing for failure may seem a bit redundant if the `equals` calls behave as expected, but we don't know that. WDYT?

Reply:
I agree, it makes sense to add tests for failure.